pypesto/ensemble/utils.py

```diff
@@ -57,6 +59,55 @@ def read_from_csv(path: str,
                         upper_bound=upper_bound)
 
 
+def read_ensemble_from_hdf5(filename: str,
+                            type: str = OPTIMIZATION,
+                            remove_burn_in: bool = True,
+                            chain_slice: slice = None,
+                            cutoff: float = None,
+                            max_size: int = None
+                            ):
+    """
+    Function for creating an ensemble from hdf5 storage file.
+
+    Parameters:
+    ----------
+    filename:
+        Name or path of the hdf5 file.
+    type:
+        Which type of ensemble to create. From History, from
+        Optimization or from Sample.
+
+    Returns:
+    -------
+    ensemble:
+        Ensemble object of parameter vectors
+    """
+    # TODO: add option HISTORY. Need to fix
+    #  reading history from hdf5.
+    if type == OPTIMIZATION:
+        if cutoff is None or max_size is None:
+            raise TypeError('You need to define cutoff and max_size '
+                            'for the ensemble creation to work.')
+        result = read_result(filename=filename,
+                             profile=False,
+                             sample=False)
```

Comment:
It was meant to keep Memory usage lower by only loading what is needed. May be unnecessary as it should get removed after the function but still should save time.
Would say as long as we have only the three (optimise, profile, sample) I would keep it, as you would have to set two values in either case (e.g. `sample = True, all = False` in your example and `profile=False, optimize=False`) when only loading one, but it you would have to set three if you wanted to save two in your case. But good to keep in mind 👍

Reply:
Hm agreed, could be
```python
def read_result(
    filename: str,
    optimize: bool = False,
    profile: bool = False,
    sample: bool = False,
):
    if not any([optimize, profile, sample]):
        optimize = True
        profile = True
        sample = True

    # read optimize
    # read profile
    # read sample
```

Then everything is loaded unless the user specifies something in particular.

Or use the `all` argument above, but set its default to `False`, then the user must always specify what to load otherwise nothing is loaded.

I would prefer one of these two options (in particular the option without `all`)

Reply:
I like that one, will go for it. Should I do it in this PR or separately as it is quite a different topic (not sure on how to handle that best).

Reply:
merged the PR 👍